agent/wpthook/test_state.cc

```diff
@@ -319,7 +325,16 @@ bool TestState::IsDone() {
   if (_active) {
     bool is_page_done = false;
     CString done_reason;
-    if (test_ms >= _test._minimum_duration) {
+    if (_test._enable_per_test_timeout && 
+        _test._timeout_value_in_seconds != 0 && 
+        ElapsedMsFromTimeoutStart(now) > 
+        _test._timeout_value_in_seconds * 1000) {

```

Comment:
This condition is a tad hard to read. I'd suggest saving ElapsedMsFromTimeoutStart(now) into a temp var. Also, if you agree to my previous comment, you don't need _enable_per_test_timeout. In short, it'll look like this: 

```c
DWORD elapsed_timeout_ms = ElapsedMsFromTimeoutStart(now);
if (_test._timeout_value_in_seconds != 0 &&
  elapsed_timeout_ms > _test._timeout_value_in_seconds * 1000)
```